main.go

```diff
@@ -0,0 +1,517 @@
+// evaluation, using data from an existing PostgreSQL database. See README.md
+// for details.
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	sp "cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	instance "cloud.google.com/go/spanner/admin/instance/apiv1"
+	"google.golang.org/api/iterator"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+	instancepb "google.golang.org/genproto/googleapis/spanner/admin/instance/v1"
+
+	"harbourbridge/internal"
+	"harbourbridge/internal/postgres"
+	"harbourbridge/spanner"
+	"harbourbridge/spanner/ddl"
+)
+
+var (
+	badDataFile      = "dropped.txt"
+	schemaFile       = "schema.txt"
+	reportFile       = "report.txt"
+	dbNameOverride   string
+	instanceOverride string
+	filePrefix       = ""
+	verbose          bool
+)
+
+func main() {
+	flag.Usage = usage
+	flag.Parse()
+	internal.VerboseInit(verbose)
+	now := time.Now()
+	lf, err := setupLogFile()
+	if err != nil {
+		fmt.Printf("\nCan't set up log file: %v\n", err)
+		panic(fmt.Errorf("can't set up log file"))
+	}
+	defer close(lf)
+	project, err := getProject()
+	if err != nil {
+		fmt.Printf("\nCan't get project: %v\n", err)
+		panic(fmt.Errorf("can't get project"))
+	}
+	instance, err := getInstance(project)
+	if err != nil {
+		fmt.Printf("\nCan't get instance: %v\n", err)
+		panic(fmt.Errorf("can't get instance"))
+	}
+	fmt.Printf("WARNING: Please check that permissions for this instance are appropriate.\n")
+	fmt.Printf("Spanner manages access control at the database level, and the database\n")
+	fmt.Printf("created by HarbourBridge will inherit default permissions from this instance.\n")
+	fmt.Printf("All data written to Spanner will be visible to anyone who can access the\n")
+	fmt.Printf("created database. Note that PostgreSQL table-level and row-level ACLs\n")
+	fmt.Printf("are dropped during conversion since they are not supported by Spanner.\n")
+	f, n, err := getSeekable(os.Stdin)
+	if err != nil {
+		fmt.Printf("\nCan't get seekable input file: %v\n", err)
+		panic(fmt.Errorf("can't get seekable input file"))
+	}
+	defer f.Close()
+	conv := postgres.MakeConv()
+	dbName, err := getDatabaseName(now)
+	if err != nil {
+		fmt.Printf("\nCan't get database name: %v\n", err)
+		panic(fmt.Errorf("can't get database name"))
+	}
+	setFilePrefix(dbName)
+	firstPass(f, n, conv)
+	writeSchemaFile(conv, now, filePrefix+schemaFile)
+	db, err := createDatabase(project, instance, dbName, now, conv)
+	if err != nil {
+		fmt.Printf("\nCan't create database: %v\n", err)
+		panic(fmt.Errorf("can't create database"))
+	}
+	client, err := getClient(db)
+	if err != nil {
+		fmt.Printf("\nCan't create client for db %s: %v\n", db, err)
+		panic(fmt.Errorf("can't create Spanner client"))
+	}
+	_, err = f.Seek(0, 0)
```

Comment:
Hmm. I don't think so.
This is part of the setup for the second pass, so logically it belongs in this chunk of code.

Moreover, this specific check is very unlikely to fail since we've already checked that the file is seekable and we've already read the entire file. That pretty much leaves file system corruption issues, or os issues as likely causes, which a very unlikely.